src/Workspaces/Core/Portable/FindSymbols/SyntaxTree/SyntaxTreeIndex.cs

```diff
@@ -58,24 +62,31 @@ public static async Task PrecalculateAsync(Document document, CancellationToken
 
         public static async Task<SyntaxTreeIndex> GetIndexAsync(
             Document document,
+            bool loadOnly,
             CancellationToken cancellationToken)
         {
             // See if we already cached an index with this direct document index.  If so we can just
             // return it with no additional work.
             if (!s_documentToIndex.TryGetValue(document, out var index))
             {
-                index = await GetIndexWorkerAsync(document, cancellationToken).ConfigureAwait(false);
+                index = await GetIndexWorkerAsync(document, loadOnly, cancellationToken).ConfigureAwait(false);
+                Contract.ThrowIfFalse(index != null || loadOnly == true, "Result can only be null if 'loadOnly: true' was passed.");
 
-                // Populate our caches with this data.
+                if (index == null && loadOnly)
+                {
+                    return null;
+                }
+
+                // Populate our cache with this data.
                 s_documentToIndex.GetValue(document, _ => index);
-                s_documentIdToIndex.GetValue(document.Id, _ => index);
```

Comment:
i would expect the fix to go here.

Reply:
can you move `s_documentToIndex.GetValue(document, _ => index) ` to be next to the `s_documentIdToIndex` updating code?  either move this code there, or move the other code here.

Reply:
fixed

Reply:
#resolved